code/math/curve.h

```diff
@@ -37,12 +39,16 @@ public :
 	//Get
 	float GetValue(float x_val) const;
 
+	// Get
+	float GetValueIntegrated(float x_val) const;
+
 	//Set
 	void ParseData();
 };
 
 extern SCP_vector<Curve> Curves;
 
 extern int curve_get_by_name(const SCP_string& in_name);
+extern int pdf_to_cdf(int curve);
```

Comment:
Statically converting the curves into their cdf versions ahead of time, which you decided against, isn't a bad idea. Why not?

Reply:
Okay. So, I mostly followed @BMagnu's lead on this PR, because he actually understands the math. But the argument against static conversion, as I remember it, went something like this:

In order to get an inverted CDF from a PDF, we have to integrate and then invert. Integration we can do, but inversion we basically can't. The binary search approach..._somehow_ lets us skip the inversion step.

I think there was also something about elegance concerns? Though of course that's more subjective.

Reply:
Okay, commentary from BMagnu that I will just copypaste here:

> TL;DR for that:
> Curves are currently not capable of representing an arbitrary integral of the input. Notably the Circular keyframe type requires stuff that curves just can't represent when integrated. And interpolating using linear functions seems just plain ugly if you can get an exact integral as well. Plus, it's really not that expensive. Especially for curves with 2 keyframes, it's actually identical to evaluating a normal curve, and beyond that, it's not that much more expensive that it's actually any sort of issue.

Reply:
Ah yes, that's right, I forgot about the need for inversion.